core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -179,11 +180,57 @@ public InternalEngine(EngineConfig engineConfig, boolean skipInitialTranslogReco
         logger.trace("created new InternalEngine");
     }
 
-    private Translog openTranslog(EngineConfig engineConfig, IndexWriter writer, boolean createNew) throws IOException {
+    @Override
+    public InternalEngine recoverFromTranslog() throws IOException {
+        flushLock.lock();
+        try (ReleasableLock lock = readLock.acquire()) {
+            ensureOpen();
+            if (openMode != EngineConfig.OpenMode.OPEN_INDEX_AND_TRANSLOG) {
+                throw new IllegalStateException("Can't recover from translog with open mode: " + openMode);
+            }
+            if (allowCommits.get()) {
+                throw new IllegalStateException("Engine has already been recovered");
+            }
+            try {
+                recoverFromTranslog(engineConfig.getTranslogRecoveryPerformer());
+            } catch (Throwable t) {
+                allowCommits.set(false); // just play safe and never allow commits on this
```

Comment:
I wanted to remove the `allowCommit.set(false)` here with an ensureOpen at the beginning of the method. Only saw later it's already there. No doubles.